src/breedsDb.js

```diff
@@ -18,13 +18,63 @@ class BreedRepository {
         return this.breeds;
     }
 
+    set setBreeds(breedsCopy) {
+        this.breeds = breedsCopy;
+    }
+
+    findIndexById(id) {
+        const breedsCopy = this.getAllBreeds; 
+        const foundBreed = breedsCopy.findIndex((breed) => {
+            return breed.breedId === id;       
+        });
+        return foundBreed;  // foundBreed is either -1 or a valid index
+    }
+
     getBreedById (id) {
         const breedsCopy = this.getAllBreeds; 
-        const idMatchArray = breedsCopy.filter((breed) => { 
+        const idMatchArray = breedsCopy.filter((breed) => {  // could use findIndexById instead of array.filter
             return breed['breedId'] == id; 
         })
         return idMatchArray[0];
     }
+
+    updateBreedById (id, object) {
+        const breedsCopy = this.getAllBreeds; 
```

Comment:
I think this might be a bug, as it's not invoking the function? Or perhaps my JS syntax is off..